samtranslator/model/eventsources/push.py

```diff
@@ -921,7 +924,29 @@ def _add_swagger_integration(  # type: ignore[no-untyped-def] # noqa: too-many-s
                 path=self.Path, method_name=self.Method, request_parameters=parameters
             )
 
-        api["DefinitionBody"] = editor.swagger
+        if api.get("MergeDefinitions", False):
+            self._merge_definitions(api, editor.swagger)
+        else:
+            api["DefinitionBody"] = editor.swagger
+
+    def _merge_definitions(self, api: Dict[str, Any], generated_body: Dict[str, Any]) -> None:
```

Comment:
nit: I slightly prefer have it returns a new merged definition instead of mutating the input.

Reply:
Updated